grpcutil/grpc.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package grpcutil provides utilities for getting information from the grpc context.
+package grpcutil
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/abcxyz/pkg/jwtutil"
+	grpcmetadata "google.golang.org/grpc/metadata"
+)
+
+const IAMKeyEndpoint = "https://www.googleapis.com/oauth2/v3/certs"
+
+// GRPCAuthenticationHandler allows for retrieving principal information from JWT tokens stored in GRPC metadata.
+type GRPCAuthenticationHandler struct {
```

Comment:
In the new location, a better name would be something like `JWTAuthenticator`. GRPC would be redundant. And JWT might be one way to authenticate.